test/unit/block_test.rb

```diff
@@ -8,4 +8,12 @@ def test_no_allocation_of_trimmed_strings
     template = Liquid::Template.parse("{{ -}} foo {{- }}")
     assert_equal 3, template.root.nodelist.size
   end
+
+  def test_pre_trim
+    template = Liquid::Template.parse("\n{%- raw %}{% endraw %}")
+    assert_equal "", template.render
+
+    template = Liquid::Template.parse("\n{%- if true %}{% endif %}")
+    assert_equal "", template.render
+  end
```

Comment:
This should really be a part of the liquid integration tests, since it isn't liquid-c specific